### PR TITLE
Action class completely tested

### PR DESCRIPTION
**Pull request by karelvandoesburg**

### CODE REVIEW

File: src/main/java/tetris/Action.java [hunk 1]
```diff
@@ -121,6 +121,7 @@ public void attempt(AbstractTetromino tetromino, Grid grid) {
     INVALID_ACTION {
         @Override
         public void attempt(AbstractTetromino tetromino, Grid grid) {
+
```
Comment (bstinenbosch) [inline src/main/java/tetris/Action.java]: Please remove white space

File: src/test/java/tetris/ActionTest.java [hunk 2]
```diff
@@ -1,69 +1,107 @@
...
+@SuppressWarnings("deprecation")
 public class ActionTest {
     @Test
     public void testChecksoftDrop() {
         DummyController controller = new DummyController();
-        Grid grid = new Grid(controller, 2, 2);
-        DummyShapeO tetromino = new DummyShapeO(new Coordinate(0, 0));
+        Grid grid = new Grid(controller, 10, 20);
+        DummyShapeO tetromino = new DummyShapeO(new Coordinate(5, 0));
         Action.SOFT_DROP.attempt(tetromino, grid);
```
Comment (bstinenbosch) [inline src/test/java/tetris/ActionTest.java]: How are you testing the attempt methods?

Comment (karelvandoesburg) [inline src/test/java/tetris/ActionTest.java]: I just run then in different situations, in one where the action is possible and in one where the action is not, where it therefore has to follow the second procedure. Maybe it's an idea to get the coordinates and see if their in the right place?

**Comment by RobertLevenbach** [comment]
There are conflicts, could you check these before I'll merge

File: src/test/java/tetris/ActionTest.java [hunk 3]
```diff
@@ -1,69 +1,136 @@
...
         DummyController controller = new DummyController();
-        Grid grid = new Grid(controller, 2, 2);
-        DummyShapeI tetromino = new DummyShapeI(new Coordinate(10, 6));
+        Grid grid = new Grid(controller, 10, 20);
+        DummyShapeJ tetromino = new DummyShapeJ(new Coordinate(9, 5));
         Action.ROTATE_RIGHT.attempt(tetromino, grid);
+        assertTrue(tetromino.left() == 7);
+    }
+
+    @Test
```
Comment (bstinenbosch) [inline src/test/java/tetris/ActionTest.java]: i'm missing a assert here.

File: src/test/java/tetris/SettingsTest.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package tetris;
+
+import org.junit.Test;
+
+public class SettingsTest {
+
+    @Test
```
Comment (bstinenbosch) [inline src/test/java/tetris/SettingsTest.java]: already tested in master, and no assertion. please remove this test.

File: src/test/java/tetris/SettingsTest.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    public void testSaveAndLoadSettings() {
+        Settings settings = new Settings();
+        settings.saveSettings();
+    }
+
+    @Test
```
Comment (bstinenbosch) [inline src/test/java/tetris/SettingsTest.java]: Same here. You can remove your settings test.

**Comment by RobertLevenbach** [comment]
looks good merging!

**Pull request merged**